datadog_checks_dev/datadog_checks/dev/tooling/commands/test.py

```diff
@@ -217,7 +217,9 @@ def test(
                 if errors:
                     abort(f'\nError collecting base package dependencies: {errors}')
 
-                spec_set = list(check_base_dependencies['datadog-checks-base'].keys())[0]
+                spec_set = list(check_base_dependencies['datadog-checks-base'].keys())
+                if len(spec_set) >= 1:
+                    spec_set = spec_set[0]
```

Comment:
(?) `spec_set` needs to remain a collection. Right now `get_next()` would probably fail since it expects an iterable.

```suggestion
                    spec_set = [spec_set[0]]
```

Also, don't we actually want to fail if no spec was found? I assume we should put a min pin on all integrations, like this PR now does?

Reply:
Hmm, this should have been collected as an error earlier from the `read_check_base_dependencies`, there must be a logic problem upstream.

`spec_set` is itself a type of collection, so no need to wrap it.

Reply:
Ahh, this is failing for the test of `datadog_checks_base` which would naturally have no dependency on itself and it explicitly gets skipped in `read_check_base_dependencies`: https://github.com/DataDog/integrations-core/blob/fd5df6b1d683ed04f1e90c54e873617274c44fa5/datadog_checks_dev/datadog_checks/dev/tooling/dependencies.py#L88

Reply:
> Also, don't we actually want to fail if no spec was found? I assume we should put a min pin on all integrations, like this PR now does?
 The build fails if there is not number at all and it would fail if the number is not high enough so I don't think we should enforce a particular number